components/match2/commons/match_group_input.lua

```diff
@@ -583,7 +654,161 @@ function MatchGroupInput.readMvp(match)
+---@param manualPlayersInput {players: table[], substitutions: MatchGroupInputSubstituteInformation[]}
+---@param options readOpponentOptions
+---@param dateTimeInfo {timestamp: integer?, timezoneOffset: string?}
+---@return standardPlayer[]
+---@return table
+function MatchGroupInput.readPlayersOfTeamNew(opponent, manualPlayersInput, options, dateTimeInfo)
+	---@type string
+	local teamName = opponent.name
+
+	local players = {}
+	local playersIndex = 0
+
+	local insertIntoPlayers = function(player)
+		if type(player) ~= 'table' or Logic.isEmpty(player) or Logic.isEmpty(player.name or player.pageName) then
+			return
+		end
+
+		local pageName = Logic.nilIfEmpty(player.name) or player.pageName
+		pageName = options.resolveRedirect and mw.ext.TeamLiquidIntegration.resolve_redirect(pageName) or pageName
+		local normalizedPageName = pageName:gsub(' ', '_')
+
+		playersIndex = playersIndex + 1
+		players[normalizedPageName] = Table.merge(players[normalizedPageName] or {}, {
+			pageName = pageName,
+			flag = Flags.CountryName(player.flag),
+			displayName = Logic.nilIfEmpty(player.displayname) or player.displayName,
+			faction = player.faction and Faction.read(player.faction) or nil,
+			index = playersIndex,
+		})
+	end
+
+	local playerIndex = 1
+	local varPrefix = teamName .. '_p' .. playerIndex
+	local name = Variables.varDefault(varPrefix)
+	-- if we do not find a player for the teamName try to find them for the teamName with underscores
+	if not name then
+		teamName = teamName:gsub(' ', '_')
+		varPrefix = teamName .. '_p' .. playerIndex
+	end
+
+	while name do
+		if options.maxNumPlayers and (playersIndex >= options.maxNumPlayers) then break end
+
+		local wasPresentInMatch = function()
+			if not dateTimeInfo.timestamp then return true end
+
+			local joinDate = DateExt.readTimestamp(Variables.varDefault(varPrefix .. 'joindate', ''))
+			local leaveDate = DateExt.readTimestamp(Variables.varDefault(varPrefix .. 'leavedate', ''))
+
+			if (not joinDate) and (not leaveDate) then return true end
+
+			-- need to offset match time to correct timezone as transfers do not have a time associated with them
+			local timestampLocal = dateTimeInfo.timestamp + DateExt.getOffsetSeconds(dateTimeInfo.timezoneOffset or '')
+
+			return (not joinDate or (joinDate <= timestampLocal)) and
+				(not leaveDate or (leaveDate > timestampLocal))
+		end
+
+		if wasPresentInMatch() then
+			insertIntoPlayers{
+				pageName = name,
+				displayName = Variables.varDefault(varPrefix .. 'dn'),
+				flag = Variables.varDefault(varPrefix .. 'flag'),
+				faction = Variables.varDefault(varPrefix .. 'faction'),
+			}
+		end
+		playerIndex = playerIndex + 1
+		varPrefix = teamName .. '_p' .. playerIndex
+		name = Variables.varDefault(varPrefix)
+	end
+
+	Array.forEach(manualPlayersInput.players, insertIntoPlayers)
+
+	--handle `substitutes` input for opponenets
+	Array.forEach(manualPlayersInput.substitutions, function(sub)
+		if sub.player and not sub.games then
```

Comment:
Not sure I get the not sub.games conidition?

Reply:
the input indicates that the sub only played in certain games/maps of the match
hence the player pushed into players table can stay since he still played in some of the games of the match